src/EditorFeatures/Core/Shared/Options/FeatureOnOffOptions.cs

```diff
@@ -78,6 +78,14 @@ internal static class FeatureOnOffOptions
 
         public static readonly PerLanguageOption<bool> StreamingGoToImplementation = new PerLanguageOption<bool>(
             nameof(FeatureOnOffOptions), nameof(StreamingGoToImplementation), defaultValue: true);
+
+        public static readonly Option<bool> NavigateToDecompiledSources = new Option<bool>(
+            nameof(FeatureOnOffOptions), nameof(NavigateToDecompiledSources), defaultValue: false,
+            storageLocations: new RoamingProfileStorageLocation($"TextEditor.{nameof(NavigateToDecompiledSources)}"));
+
+        // Note: no storage location since this is intentionally a session variable
```

Comment:
It might be worth explaining why this is.  A session variable actually seems pretty unpleasant to me.  Especially because it's paired with a model dialog box.  If i have to dismiss this modal dialog each and every session i navigate to decompiled code, i'm going to get very annoyed very fast.  

It seems odd given that i can't think of any other legal disclaimer in any other app i've used (MS or otherwise) that requires approval each and every session.  All the ones i can think of force me to explicit accept once, but then seem to persist forever (or until something like a new policy change comes through).

Reply:
➡️ Discussions are already in progress to reduce the number of times this is shown.